V6.0.2

DustinGrau opened the pull request:
Updates to the latest 3rd-party libraries for builds, bringing devices up to date with the latest software features. Addresses all minor issues and fixes discovered since the v5.4.4 and v6.0 releases. Also includes some housekeeping to rename items for more clarity and consistency with documentation, while fixing said content to remain in agreement and avoid confusion.

Resolved Issues:
#426
#442
#449
#451

Ghosts Busted:
-Fixed bluetooth radio accidentally being enabled on Attenuator/Wifi
-Fixed incorrect power level segment spacing on 30-segment bargraph
-Fixed issue where AL/FE beep loop could double when rapidly changing power level
-Fixed issue where clearing the EEPROM settings on the pack or wand would not always clear settings
-Fixed issue where custom bargraph firing animation EEPROM setting would sometimes be ignored
-Fixed issue where music of specific length could cause auto-skip to fail
-Fixed missing board status LED on single-shot blaster during POST
-Fixed occasional stuck segment issue when ramping down 28-segment bargraph
-Fixed web UI occasionally displaying stale data if pack desync occurs
-Added RGB LED white balancing
-Adjusted AL/FE beep loops to better match on-screen sound and timing
-Disabled AL/FE beep loops while firing to better match on-screen sound
-Disabled RGB powercell by default to better match TVG
-Disabled vibration motor on standalone wands to improve power consumption
-Disabled wand boot errors by default to better match films
-Made power level 5 the default startup power level to better match films
-Reduced automatic music track skip delay by 50%
-Reduced max speaker output on standalone wands to improve power consumption
-Various "under the hood" fixes and optimizations

New Features:
-Added "hot shutdown" feature to allow turning off wand without turning off pack
-Added music playback status to web UI
-Made 84/89 cyclotron cycle rate dynamic based on whether LED fading is enabled or disabled
-Made it so wand will automatically power on in demo light mode if Activate switch is on
-Made RGB inner cyclotron panel the default setting
-Updated all firmware flash tools
-Updated documentation
-Updated STLs

DustinGrau committed:
Update to latest libraries, merge minor fixes

DustinGrau committed:
Merge more changes from 6.1

DustinGrau committed:
Sync changes to branch

DustinGrau committed:
Sync changes from 6.1 dev

DustinGrau committed:
Update BELT_GIZMO.md

DustinGrau committed:
Fix wand benchtest flags for compile

DustinGrau committed:
Adjust docs, scripts

DustinGrau committed:
Update libraries, recompile

DustinGrau committed:
Update libraries, recompile

DustinGrau commented:
Making this PR ready for review, though I have yet to put the updated binaries into any of my actual equipment, so testing is DEFINITELY needed.

DustinGrau committed:
Reset benchtest flag on Wand

DustinGrau committed:
Fix compile script, update binaries

nomakewan committed:
Update docs, defaults

Updates Attenuator documentation to account for the updated command syntax for esptool, reverts RGB vent light defaults

nomakewan committed:
Vibration, startup fixes

Reverts vibration changes, enforces no vibration by default on standalone wand, adds ability to keep pack running when wand is shut down if vent switch is left on when activate switch is turned off in Mode Super Hero, adds wand-only startup to Afterlife and Frozen Empire.

nomakewan committed:
Typo fix

nomakewan reviewed (approved):
Okay, now that I've had a chance to test the code I've fixed the issues I found, namely that the vibration motor intensity reduction was causing the motor to stall briefly on initial start due to insufficient momentum, which was counterintuitively causing higher current draw than if it were just active at a higher duty cycle.

The vibration feature has now been disabled by default on standalone wands (though it can be turned back on in the EEPROM configuration menu if a user desires), and a new "hot shutdown" feature has been added to address community requests to implement a way to shut down the wand without shutting down the pack.

Looks good on my end!

nomakewan committed:
Update ADVCONFIG.md

Fixes default for wand boot errors

nomakewan committed:
Hot boot SFX fix

Fixes order-of-operations issue that would cause erroneous doubling of bootup sounds if Proton Pack were turned off when wand is booted with all switches on in Mode Super Hero.

nomakewan committed:
Update default power level

Changes default power level for the wand and pack to be level 5 to better match on-screen behavior, updates beep loops to better match on-screen audio tone and length.

nomakewan committed:
Adjust beep volume

Adjusts volume level of new beeps to better blend with AL/FE idle loops

nomakewan committed:
Update 84/89 cycle rate

Changes 84/89 cycle rate to be dynamic based on whether the user has enabled or disabled LED fading; LED fading mimics TVG, while fading disabled mimics GB1/GB2.

nomakewan committed:
Code cleanup, docs update

Cleans up the new dynamic 84 animation code to make it easier to maintain, updates documentation

nomakewan committed:
Disable beep loop while firing

This disables the beep loop in AL/FE while firing.

DustinGrau committed:
Update OPERATION_USAGE.md

DustinGrau committed:
Merge branch 'v6.0.Fixes' of https://github.com/gpstar81/GPStar-proton-pack into v6.0.Fixes

DustinGrau committed:
Doc updates

nomakewan committed:
Fix firing beep logic

Fixes potential for overlapping beeps when firing stops by moving the beep loop stop control into the main firing loop so that it runs constantly just like in the main loop code.

nomakewan committed:
Demo mode code cleanup

Cleanup for demo light mode pack code for efficiency

nomakewan committed:
Fix build scripts to match 6.0.x

Adjusts build scripts to match layout and contents expected by the 6.0.x builds.

nomakewan committed:
Update gpstarFirmwareFlasher.exe

Adds support for Arduino LLC Mega 2560 USB, Arduino LLC Mega 2560 R3 USB, and Mega 2560 Pro boards with CH340G bridge chips.

nomakewan committed:
Update pioarduino

Updates pioarduino dependency for ESP32 builds to latest release

nomakewan committed:
Fix EEPROM clear functionality

Fixes range error in EEPROM clear functionality that would cause an EEPROM reset to fail to actually reset the EEPROM.

nomakewan committed:
Bargraph i2c refactor

Since the HT16K33 can only ever be detected on 0x70, this changes the i2c scan routine on all devices with a bargraph to only test that address for a response rather than 1-126.

nomakewan commented:
Testing looks good on my end; that should do it for changes related to ongoing work in 6.1.Dev as well as user bug reports. LGTM!

DustinGrau commented:
I just did a full update and gave it a go, went through all the streams on every mode. Made sure the button-mash behavior did what it should (notably for GB:FE) and doing so with and without the lid on the cyclotron. Best I can tell all the usual behavior is there and working, so let's hope we fixed more bugs than we possibly will have caused :)
